### PR TITLE
[cssom] Add test disallowing non-standard feature

### PR DESCRIPTION
**Pull request by jugglinmike**
Safari 12 on macOS Mojave currently fails all but the `radio` and `checkbox` sub-tests. As noted in the referenced bug, the WebKit maintainers intend to release a fix by September of this year.

### CODE REVIEW

File: css/cssom/cssstyledeclaration-setter-form-controls.html
Comment (zcorpan) [inline css/cssom/cssstyledeclaration-setter-form-controls.html]: The tests are very similar, only varying on the element and attribute. Can you make it more DRY?

For example,
```
const tests = [['input', 'text'], ..., ['button']];
for (const t of tests) {
  var [tagName, type] = t;
  test(...);
}
```

**Review by zcorpan — CHANGES_REQUESTED** [review]

**Comment by jugglinmike** [comment]
I'd rather not use control flow structures to define or execute tests. They can hide test bugs and make it harder for others to understand what's going on. But that doesn't mean we can't reduce repetition! I've refactored to use a few helper functions while keeping the tests declarative.

**Review by zcorpan — APPROVED** [review]

**Comment by jugglinmike** [comment]
@zcorpan would you like me to receive review from any other maintainers before merging this?

**Comment by zcorpan** [comment]
Nope, I can merge.

**Pull request merged**
